### PR TITLE
19 limit api

### PR DESCRIPTION
**Pull request by sinarani**
added auth and limit api

### CODE REVIEW

File: expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Limit limit=limitDao.findById(limitModel.getId()).orElseThrow(NoSuchElementException::new);
+            limitModel.setId(limit.getId());
+          limit1=  limitDao.save(limitConversion.getEntity(limitModel));
+
+        return limitConversion.getModel(limit1);
+    }
+
+    @Override
+    public void deleteLimit(long id, Principal principal) {
+        Limit limit=limitDao.findById(id).orElseThrow(NoSuchElementException::new);
```
Comment (codesnaper) [inline expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java:66]: Check with respect to user id also

File: expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Limit limit1 = null;
+        Limit limit=limitDao.findById(limitModel.getId()).orElseThrow(NoSuchElementException::new);
+            limitModel.setId(limit.getId());
+          limit1=  limitDao.save(limitConversion.getEntity(limitModel));
+
+        return limitConversion.getModel(limit1);
+    }
+
+    @Override
+    public void deleteLimit(long id, Principal principal) {
```
Comment (codesnaper) [inline expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java]: Pass string user id

File: expense-api/src/main/java/com/expense/expensemanagement/service/category/CategoryService.java [hunk 3]
```diff
@@ -45,23 +47,20 @@ public com.expense.expensemanagement.model.Category addCategory(com.expense.expe
...
 
-
-        return categoryDto1;
     }
 
     @Override
-    public void deleteCategory(long id) {
-
-        categoryDao.deleteById(id);
+    public void deleteCategory(long id, Principal principal) {
```
Comment (codesnaper) [inline expense-api/src/main/java/com/expense/expensemanagement/service/category/CategoryService.java]: pass user id as parameter

File: expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java (hunk 1 above)
Comment (sinarani) [reply expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java:66]: updated

File: expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java (hunk 2 above)
Comment (sinarani) [reply expense-api/src/main/java/com/expense/expensemanagement/service/limit/LimitService.java]: updated

File: expense-api/src/main/java/com/expense/expensemanagement/service/category/CategoryService.java (hunk 3 above)
Comment (sinarani) [reply expense-api/src/main/java/com/expense/expensemanagement/service/category/CategoryService.java]: updated

**Review by codesnaper — APPROVED** [review]

**Pull request merged**